### ISSUE TITLE
Assert::isMap in HtpasswdFactory leads to Exception: 'Expected map - associative array with string keys.'

### ISSUE THREAD

**Issue by seboettg** [mezzio/mezzio-authentication#10]
### Bug Report

|    Q        |   A
|------------ | ------
| Version(s)  | 1.2.0

#### Summary

After running `composer update` HtpasswdFactory does not work anymore. While debugging I could figure out that `Assert::isMap($config)` leads to a `InvalidArgumentException` with message 'Expected map - associative array with string keys.'
When I comment this line everything works as excepted.

#### Current behavior

Application terminates with Error 500 caused by InvalidArgumentException

#### How to reproduce

I don't know...

my config looks like this:

```
[
    'authentication' => [
        'realm' => 'doc',
        'htpasswd' => __DIR__ . "/../config/htpasswd",
    ]
]
```

#### Expected behavior

Htpasswd object will be returned.

**Comment by Ocramius** [comment]
@seboettg what's actually passed to the `Assert::isMap()` call?

**Comment by seboettg** [comment]
Hi @Ocramius!
Thank you for your quick answer. The passed `$config` is an ArrayObject.

**Comment by boesing** [comment]
> The passed $config is an ArrayObject.

hmpf, I remember that there was some kind of "compatibility" reason within mezzio to allow other containers like symfony container. Thus, the config is being converted to a recursive array object afair...

https://github.com/jsoumelidis/zend-sf-di-config/blob/master/src/Config.php#L88 for example
https://github.com/laminas/laminas-auradi-config/blob/2.1.x/src/Config.php#L66
[...]

**Comment by boesing** [comment]
Closed with #11

**Comment by seboettg** [comment]
Wow, that went very fast. Thank you.
